tests_metricflow/snapshots/test_name_edge_caes.py/SqlPlan/DuckDB/test_filter_by_metric_name_with_2_measures_from_same_source_node__plan0.sql

```diff
@@ -6,369 +6,283 @@ sql_engine: DuckDB
+                  , EXTRACT(month FROM bookings_source_src_32000.ds) AS booking__ds__extract_month
+                  , EXTRACT(day FROM bookings_source_src_32000.ds) AS booking__ds__extract_day
+                  , EXTRACT(isodow FROM bookings_source_src_32000.ds) AS booking__ds__extract_dow
+                  , EXTRACT(doy FROM bookings_source_src_32000.ds) AS booking__ds__extract_doy
+                  , 'dummy_dimension_value' AS homonymous_metric_and_entity__homonymous_metric_and_dimension
+                  , DATE_TRUNC('day', bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__day
+                  , DATE_TRUNC('week', bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__week
+                  , DATE_TRUNC('month', bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__month
+                  , DATE_TRUNC('quarter', bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__quarter
+                  , DATE_TRUNC('year', bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__year
+                  , EXTRACT(year FROM bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__extract_year
+                  , EXTRACT(quarter FROM bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__extract_quarter
+                  , EXTRACT(month FROM bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__extract_month
+                  , EXTRACT(day FROM bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__extract_day
+                  , EXTRACT(isodow FROM bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__extract_dow
+                  , EXTRACT(doy FROM bookings_source_src_32000.ds) AS homonymous_metric_and_entity__ds__extract_doy
+                  , 'dummy_entity_value' AS homonymous_metric_and_entity
+                  , 'dummy_entity_value' AS booking__homonymous_metric_and_entity
+                FROM ***************************.fct_bookings bookings_source_src_32000
+              ) subq_0
+            ) subq_1
+          ) subq_2
           WHERE homonymous_metric_and_entity IS NOT NULL
-        ) subq_2
-      ) subq_3
+        ) subq_3
+      ) subq_4
       GROUP BY
-        subq_3.metric_time__day
-    ) subq_4
-  ) subq_5
+        subq_4.metric_time__day
+    ) subq_5
+  ) subq_6
   FULL OUTER JOIN (
     -- Compute Metrics via Expressions
     SELECT
-      subq_9.metric_time__day
-      , subq_9.__homonymous_metric_and_dimension AS homonymous_metric_and_dimension
+      subq_11.metric_time__day
+      , subq_11.__homonymous_metric_and_dimension AS homonymous_metric_and_dimension
     FROM (
       -- Aggregate Inputs for Simple Metrics
       SELECT
-        subq_8.metric_time__day
-        , SUM(subq_8.__homonymous_metric_and_dimension) AS __homonymous_metric_and_dimension
+        subq_10.metric_time__day
+        , SUM(subq_10.__homonymous_metric_and_dimension) AS __homonymous_metric_and_dimension
```

Comment:
The filter `WHERE homonymous_metric_and_entity IS NOT NULL` is still invalid, but the optimized query works as kind of a fluke.

Reply:
Yeah, seems like it. The filtering behavior is odd and probably not what the user is expecting. It's acting as a filter for a measure that depends on the value of another measure.